WordPress/wordpress-develop
# `wp_theme_has_theme_json`: add public method to know whether the theme supports `theme.json`

Trac ticket https://core.trac.wordpress.org/ticket/56975

Related Gutenberg PRs:

- https://github.com/WordPress/gutenberg/pull/45168
- https://github.com/WordPress/gutenberg/pull/45380
- https://github.com/WordPress/gutenberg/pull/46150
- https://github.com/WordPress/gutenberg/pull/45979
- https://github.com/WordPress/gutenberg/pull/45950

Lnking to [Gutenberg's 6.2 backport tracking ticket](https://github.com/WordPress/gutenberg/issues/47187).

## What

Adds a new public method `wp_theme_has_theme_json` that checks whether a theme has a `theme.json` file.

## Why

Part of https://github.com/WordPress/gutenberg/issues/45171

This offers consumers a public API so they don't have to rely on private code. Additionally, it uses the object cache under the `theme_json` group to maintain this data, consolidating with the changes done for settings ([core PR](https://github.com/WordPress/wordpress-develop/pull/3712)) and settings (core PR soon, see [Gutenberg PR](https://github.com/WordPress/gutenberg/pull/45372) instead).

## How

Creates the new public method, substitutes usage, and deprecates the old one.

## Test

- Verify that automated tests pass.
- Using TwentyTwentyThree (a theme with a theme.json) and then a TwentyTwentyOne (a theme without theme.json) do the following:
  - Load the front-end. Verify that styles load as expected.
  - Load the post editor. Verify that the presets (e.g.: color) are set correctly in the block inspector.
